autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -588,6 +604,134 @@ async def create_store_submission(
         ) from e
 
 
+async def create_store_version(
+    user_id: str,
+    agent_id: str,
+    agent_version: int,
+    store_listing_id: str,
+    name: str,
+    video_url: str | None = None,
+    image_urls: list[str] = [],
+    description: str = "",
+    sub_heading: str = "",
+    categories: list[str] = [],
+    changes_summary: str = "Update Submission",
+) -> backend.server.v2.store.model.StoreSubmission:
+    """
+    Create a new version for an existing store listing
+
+    Args:
+        user_id: ID of the authenticated user submitting the version
+        agent_id: ID of the agent being submitted
+        agent_version: Version of the agent being submitted
+        store_listing_id: ID of the existing store listing
+        name: Name of the agent
+        video_url: Optional URL to video demo
+        image_urls: List of image URLs for the listing
+        description: Description of the agent
+        categories: List of categories for the agent
+        changes_summary: Summary of changes from the previous version
+
+    Returns:
+        StoreSubmission: The created store submission
+    """
+    logger.debug(
+        f"Creating new version for store listing {store_listing_id} for user {user_id}, agent {agent_id} v{agent_version}"
+    )
+
+    try:
+        # First verify the listing belongs to this user
+        listing = await prisma.models.StoreListing.prisma().find_first(
+            where=prisma.types.StoreListingWhereInput(
+                id=store_listing_id, owningUserId=user_id
+            )
+        )
+
+        if not listing:
+            logger.warning(
+                f"Store listing not found for user {user_id}: {store_listing_id}"
+            )
+            raise backend.server.v2.store.exceptions.ListingNotFoundError(
+                f"Store listing not found. User ID: {user_id}, Listing ID: {store_listing_id}"
+            )
+
+        # Verify the agent belongs to this user
+        agent = await prisma.models.AgentGraph.prisma().find_first(
+            where=prisma.types.AgentGraphWhereInput(
+                id=agent_id, version=agent_version, userId=user_id
+            )
+        )
+
+        if not agent:
+            logger.warning(
+                f"Agent not found for user {user_id}: {agent_id} v{agent_version}"
+            )
+            raise backend.server.v2.store.exceptions.AgentNotFoundError(
+                f"Agent not found for this user. User ID: {user_id}, Agent ID: {agent_id}, Version: {agent_version}"
+            )
+
+        # Get the latest version number
+        latest_version = await prisma.models.StoreListingVersion.prisma().find_first(
+            where=prisma.types.StoreListingVersionWhereInput(
+                storeListingId=store_listing_id
+            ),
+            order=[{"version": "desc"}],
+        )
```

Comment:
we can make the listing the above query to fetch this, instead of multiple calls:
```
        listing = await prisma.models.StoreListing.prisma().find_first(
            where=prisma.types.StoreListingWhereInput(
                id=store_listing_id, owningUserId=user_id
            )
            include={"Versions": { "order_by": [{"version":"desc"}], "take":  1}}
     )
```


Or even better, use max instead of fetching the whole data.

Reply:
can't find max but applied this